tools/link-verifier/verify-links.py

```diff
@@ -0,0 +1,347 @@
+    parser.add_argument("-v", "--verbose", action="store_true", default=False, help="Print all links tested")
+    args = parser.parse_args()
+    file_list = []
+    html_file_list = []
+    broken_links = []
+    if args.files is None and args.links is None:
+        parser.error('Either files or links must be provided.')
+
+    if args.files is not None:
+        # Search for markdown files in file list. Note: We could skip this if we can guarantee only markdown files are passed in.
+        file_list = [f for f in args.files if re.search(MARKDOWN_SEARCH_TERM, f, re.IGNORECASE)]
+
+        if args.verbose:
+            print(file_list)
+
+        try:
+            file_map = {}
+            for f in file_list:
+                process = create_html(f)
+                if process.returncode != 0:
+                    cprint(process.stdout, 'red')
+                    print('Did you install pandoc?')
+                    sys.exit(process.returncode)
+                html_file_list.append(html_name_from_markdown(f))
+                # Create a map so that we know what file this was generated from.
+                file_map[html_name_from_markdown(f)] = f
+
+            # Parse files in parallel.
+            pool = Pool(args.num_processes)
+            file_objects = pool.map(parse_file, html_file_list)
+            pool.close()
+            pool.join()
+            for file_obj in file_objects:
+                consolidate_repo_list(file_obj.linked_repos)
+            # Test links in series so we don't send too many HTTP requests in a short interval.
+            for file_obj in file_objects:
+                file_obj.identify_broken_links(file_map, args.verbose)
+                broken_links += file_obj.broken_links
+        # Remove the temporary files we created, especially if there was an exception.
+        finally:
+            for f in html_file_list:
+                if not args.keep:
+                    os.remove(f)
+
+    if args.links is not None:
+        for link in args.links:
+            try:
+                is_broken, status_code = test_url(link)
+                if is_broken:
+                    broken_links.append(link)
+                    cprint(f'{status_code}\t{link}', 'red')
+                else:
+                    if args.verbose:
+                        cprint(f'{status_code}\t{link}', 'green')
+            # Many things could go wrong since anything could be passed on the command line.
+            except Exception as e:
+                traceback.print_exc()
+                broken_links.append(link)
+                cprint(f'Error\t{link}', 'red')
+
+    # Return code > 0 to return error. This may return success if there are 256 broken links ¯\_(ツ)_/¯
+    num_broken = len(broken_links)
+    if num_broken > 0:
+        print(f'{num_broken} broken link' + ('s', '')[num_broken == 1])
+    sys.exit(num_broken)
```

Comment:
Can you try to avoid the overflow?
```suggestion
    if num_broken > 0:
        print(f'{num_broken} broken link' + ('s', '')[num_broken == 1])
        sys.exit(1)
    sys.exit(0)
```